core/src/test/java/com/linecorp/armeria/client/TestReconfigurableServer.java

```diff
@@ -16,50 +16,110 @@
 package com.linecorp.armeria.client;
 
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutionException;
 
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
 import org.junit.jupiter.api.Test;
 
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.server.Server;
 import com.linecorp.armeria.server.ServerBuilder;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
 class TestReconfigurableServer {
 
-    public static void main(String args[]) {
-        System.out.println("Hello....");
-    }
     @Test
-    public void test() throws InterruptedException, ExecutionException {
+    public void test_reconfiguration_of_server() throws Exception {
```

Comment:
I think we will need more unit tests and more different combinations of ServerConfig. Maybe add unit tests with ServerBuilder.routes or ServerBuilder.annotatedServices etc?